oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtIssuerAuthenticationManagerResolver.java

```diff
@@ -130,6 +165,10 @@ public AuthenticationManager resolve(HttpServletRequest request) {
 		return authenticationManager;
 	}
 
+	public void setIssuerConverter(Converter<HttpServletRequest, String> issuerConverter) {
+		this.issuerConverter = issuerConverter;
```

Comment:
Please check for `null` here using `Assert.notNull`, as you did earlier in the constructor.

Reply:
Also, I believe it would be a bit more consistent with the rest of Spring Security to call this `setIssuerResolver`. It's certainly reminiscent of `BearerTokenResolver` in its role in the request.

Reply:
Also, out of curiosity, do you have a use case where you need the more generic contract of `Converter<HttpServletRequest, String>`? If not, `setBearerTokenResolver` may be better since that's likely the most common use case.

Reply:
@jzheaux the problem is that you cannot set the `BearerTokenResolver` on the `issuerConverter` as it is of type `Converter<HttpServletRequest,String>`. So either you replace the entire thing or you would need to add a setter on that one as well and change the type of the `issuerConverter` attribute directly to `JwtClaimIssuerConverter`. What do you think?

Reply:
I agree with you that adding a setter to `JwtClaimIssuerConverter` would be one way to do it, though it would remain private.

One nice thing about using `BearerTokenResolver` is that it simplifies the API and focuses on the most common use cases. It seems like a very uncommon use case to pull the issuer from somewhere else in the request other than the token itself.